examples/alloydb-vectorstore-example/alloydb_vectorstore_example.go

```diff
@@ -0,0 +1,212 @@
+		log.Fatal("env variable ALLOYDB_DATABASE is empty")
+	}
+	// Requires environment variable PROJECT_ID to be set.
+	projectID := os.Getenv("PROJECT_ID")
+	if projectID == "" {
+		log.Fatal("env variable PROJECT_ID is empty")
+	}
+	// Requires environment variable ALLOYDB_REGION to be set.
+	region := os.Getenv("ALLOYDB_REGION")
+	if region == "" {
+		log.Fatal("env variable ALLOYDB_REGION is empty")
+	}
+	// Requires environment variable ALLOYDB_INSTANCE to be set.
+	instance := os.Getenv("ALLOYDB_INSTANCE")
+	if instance == "" {
+		log.Fatal("env variable ALLOYDB_INSTANCE is empty")
+	}
+	// Requires environment variable ALLOYDB_CLUSTER to be set.
+	cluster := os.Getenv("ALLOYDB_CLUSTER")
+	if cluster == "" {
+		log.Fatal("env variable ALLOYDB_CLUSTER is empty")
+	}
+	// Requires environment variable ALLOYDB_TABLE to be set.
+	table := os.Getenv("ALLOYDB_TABLE")
+	if table == "" {
+		log.Fatal("env variable ALLOYDB_TABLE is empty")
+	}
+
+	// Requires environment variable VERTEX_LOCATION to be set.
+	location := os.Getenv("VERTEX_LOCATION")
+	if location == "" {
+		log.Fatal("env variable VERTEX_LOCATION is empty")
+	}
+
+	return username, password, database, projectID, region, instance, cluster, table, location
+}
+
+func main() {
+	// Requires the Environment variables to be set as indicated in the getEnvVariables function.
+	username, password, database, projectID, region, instance, cluster, table, vertexLocation := getEnvVariables()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	pgEngine, err := alloydbutil.NewPostgresEngine(ctx,
+		alloydbutil.WithUser(username),
+		alloydbutil.WithPassword(password),
+		alloydbutil.WithDatabase(database),
+		alloydbutil.WithAlloyDBInstance(projectID, region, cluster, instance),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Initialize Vectorstore table using InitVectorstoreTable method
+	vectorstoreTableoptions, err := alloydbutil.NewVectorstoreTableOptions(&alloydbutil.VectorstoreTableOptions{
+		TableName:  "table",
+		VectorSize: 768,
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = pgEngine.InitVectorstoreTable(ctx, *vectorstoreTableoptions,
+		[]alloydbutil.Column{
+			alloydbutil.Column{
+				Name:     "area",
+				DataType: "int",
+				Nullable: false,
+			},
+			alloydbutil.Column{
+				Name:     "population",
+				DataType: "int",
+				Nullable: false,
+			},
+		},
+		alloydbutil.Column{Name: "langchain_id", DataType: "UUID", Nullable: false},
+		true,
+		true,
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Initialize VertexAI LLM
+	llm, err := vertex.New(ctx, vertex.WithCloudProject(projectID), vertex.WithCloudLocation(vertexLocation), vertex.WithDefaultModel("text-embedding-005"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	e, err := embeddings.NewEmbedder(llm)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new alloydb vectorstore .
+
+	vs, err := alloydb.NewVectorStore(ctx, pgEngine, e, table,
+		alloydb.WithMetadataColumns(
```

Comment:
alloydbutil.Column should be for the init table only. Please continue to use strings for the vector store options